spyderlib/spyder.py

```diff
@@ -2781,6 +2781,7 @@ def run_spyder(app, options, args):
     # Open external files with our Mac app
     if running_in_mac_app():
         app.sig_open_external_file.connect(main.open_external_file)
+        QCoreApplication.setAttribute(Qt.AA_DontShowIconsInMenus, True)

```

Comment:
Please move this line out of this `if`. `running_in_mac_app()` only tests if Spyder is running from its Mac application (i.e. the one you get when you download and install our DMG), not if Spyder is running in Mac.

What we want it this:

```python
if running_in_mac_app():
    app.sig_open_external_file.connect(main.open_external_file)

# Don't show icons in menus for Mac
if sys.platform == 'darwin':
    QCoreApplication.setAttribute(Qt.AA_DontShowIconsInMenus, True)
```

so that your fix applies to any version of Spyder installed in Mac (and not just our app ;-)